frontend/test/pytest/test_cuda_integration.py

```diff
@@ -502,5 +503,173 @@ def circuit2(a):
         assert_allclose(expected, observed)
 
 
+@pytest.mark.cuda
+class TestForLoops:
+    """Test for loops."""
+
+    def test_basic_loop(self):
+        """Test simple for loop."""
+
+        @qml.qnode(qml.device("lightning.qubit", wires=1))
+        def circuit_lightning(n):
+            @for_loop(0, n, 1)
+            def loop_fn(_):
+                qml.PauliX(0)
+
+            loop_fn()
+            return qml.state()
+
+        from catalyst.cuda import SoftwareQQPP
+
+        @qml.qnode(SoftwareQQPP(wires=1))
+        def circuit(n):
+            @for_loop(0, n, 1)
+            def loop_fn(_):
+                qml.PauliX(0)
+
+            loop_fn()
+            return qml.state()
+
+        cuda_compiled = catalyst.cuda.qjit(circuit)
+        catalyst_compiled = qjit(circuit_lightning)
+        expected = catalyst_compiled(2)
+        observed = cuda_compiled(2)
+        assert_allclose(expected, observed)
+
+    def test_loop_caried_values(self):
+        """Test for loop with updating loop carried values."""
+
+        @qml.qnode(qml.device("lightning.qubit", wires=1))
+        def circuit_lightning(n):
+            @for_loop(0, n, 1)
+            def loop_fn(_, x):
+                qml.PauliX(0)
+                return x + np.pi / 4
+
+            loop_fn(0.0)
+            return qml.expval(qml.PauliZ(0))
+
+        from catalyst.cuda import SoftwareQQPP
+
+        @qml.qnode(SoftwareQQPP(wires=1))
+        def circuit(n):
+            @for_loop(0, n, 1)
+            def loop_fn(_, x):
+                qml.PauliX(0)
+                return x + np.pi / 4
+
+            loop_fn(0.0)
+            return qml.expval(qml.PauliZ(0))
+
+        cuda_compiled = catalyst.cuda.qjit(circuit)
+        catalyst_compiled = qjit(circuit_lightning)
+        expected = catalyst_compiled(2)
+        observed = cuda_compiled(2)
+        assert_allclose(expected, observed)
+
+    def test_dynamic_wires(self):
+        """Test for loops with iteration index-dependant wires."""
+
+        @qml.qnode(qml.device("lightning.qubit", wires=6))
+        def circuit_lightning(n: int):
+            @for_loop(0, n - 1, 1)
+            def loop_fn(i):
+                qml.CNOT(wires=[i, i + 1])
+
+            loop_fn()
+            return qml.state()
+
+        from catalyst.cuda import SoftwareQQPP
+
+        @qml.qnode(SoftwareQQPP(wires=6))
```

Comment:
This is fairly minor, but we can avoid importing the device class and reference the device directly via it's short name, which is how a user would use it:
```suggestion
        @qml.qnode(qml.device("softwareq.qpp", wires=6))
```

Reply:
Yes, this is the way it should be. It was not originally the case for the rest of the tests, but it has been changed to follow this convention. Thanks for pointing it out @dime10!

Reply:
Done, changed to referencing qpp backend by name instead of import.